### DISCUSSION TITLE
Upgrade for shirou/gopsutil to v3.23.3

### DISCUSSION THREAD

**Discussion by ayeletdeRoos** [moby/moby#45444] [Q&A]
Hi, I'm using M1 MAC. the current version of the pkg shirou/gopsutil  in the docker/docker pkg (or moby/moby?) isn't supporting the m1 cpu.
it errors with:
`../../../pkg/mod/github.com/shirou/gopsutil@v0.0.0-20190901111213-e4ec7b275ada/process/process_darwin.go:591:34: undefined: KinfoProc`

it's a known issue, therefor they updated the version of the pkg to support it to version [v3.23.3](https://github.com/shirou/gopsutil/releases/tag/v3.23.3)

is there a plan to upgrade the pkg in this project?

**Comment by thaJeztah** [comment]
That package doesn't look to be a dependency of docker (moby) nor for the docker cli; neither of those contain that package in the vendor directory;

- https://github.com/docker/cli/tree/c177e2060d769e2b4527ae6d3016d2dcaa9d629b/vendor/github.com
- https://github.com/moby/moby/tree/3eca54384d77840823c650930a3fd17282a825fb/vendor/github.com

Perhaps it's an indirect dependency for something else you're using? Are you seeing this when building some package, or when running a binary?

**Reply by ayeletdeRoos** [reply]
I saw the dependency [here](https://github.com/moby/moby/blob/3eca54384d77840823c650930a3fd17282a825fb/vendor.sum#L1292)

maybe it's also indirect by you?

I'm seeing it when building a package that uses docker/docker. it can be that this dependency comes from somewhere else as well.
